pkg/scheduler/scheduler.go

```diff
@@ -84,7 +84,7 @@ type Scheduler struct {
 	schedulingCycle int64
 
 	// Stubs.
-	applyAdmission func(context.Context, *kueue.Workload) error
+	patchAdmission func(context.Context, *kueue.Workload, *kueue.Workload) error
```

Comment:
Small nit: consider adding arg names to make it clear which workload is which, i.e. (or similar):
```suggestion
	patchAdmission func(ctx context.Context, original *kueue.Workload, updated *kueue.Workload) error
```

Reply:
Or maybe I will do that as the prep PR, as reviewing this one might take a while.